cpp/src/io/json/json_column.cu

```diff
@@ -393,6 +393,41 @@ std::vector<std::string> copy_strings_to_host(device_span<SymbolT const> input,
   return to_host(d_column_names->view());
 }
 
+rmm::device_uvector<uint8_t> is_all_nulls_str_column(device_span<SymbolT const> input,
+                                                     tree_meta_t const& d_column_tree,
+                                                     tree_meta_t const& tree,
+                                                     device_span<NodeIndexT> col_ids,
+                                                     cudf::io::json_reader_options const& options,
+                                                     rmm::cuda_stream_view stream)
+{
+  auto const num_nodes = col_ids.size();
+  auto const num_cols  = d_column_tree.node_categories.size();
+  rmm::device_uvector<uint8_t> is_all_nulls(num_cols, stream);
+  thrust::fill(rmm::exec_policy(stream), is_all_nulls.begin(), is_all_nulls.end(), true);
```

Comment:
Wouldn't it be better to skip this step and just set `true` inside the for_each below?

Reply:
for_each will write only `false`, and they don't use atomics. By intializing as `true` before and writing `false` only in for_each, we could avoid atomics or `reduce_by_key` operation here.

Reply:
I don't follow. How do atomics fit in here? Are you saying that the below is being hit by multiple threads? That would be UB, even if it's only ever flipping things to false.

`if (!is_null_literal) is_all_nulls[col_ids[i]] = false;`

Reply:
Yes. multiple threads write to same location. The documentation mentioned that number of writes is undefined. 
https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#simt-architecture-notes
> If a non-atomic instruction executed by a warp writes to the same location in global or shared memory for more than one of the threads of the warp, the number of serialized writes that occur to that location varies depending on the compute capability of the device (see [Compute Capability 5.x](https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#compute-capability-5-x), [Compute Capability 6.x](https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#compute-capability-6-x), and [Compute Capability 7.x](https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#compute-capability-7-x)), and which thread performs the final write is undefined.

Reply:
No. 2 random threads could write to same location. I would like to learn the behaviour in this situation.
This is similar to setting update a cleared array of flags in global memory.

Reply:
Concurrent, non-atomic writes to the same memory location are undefined behavior. Full stop.

That said, on all current GPUs, writing the same value to the same location is _probably fine_. I'd try hard to find a different way of doing it, but if there's no better way without sacrificing significant performance, then don't lose too much sleep over it. There are far worse UB sins you could commit ( `volatile` 👀)